src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -218,18 +218,21 @@
 
   <!-- archive the test binaries along with some supporting files -->
   <Target Name="ArchiveTestBuild" AfterTargets="CreateAssemblyListTxt" Condition="'$(ArchiveTests)' == 'true'">
+    <PropertyGroup Condition="'$(TestProjectName)'==''">
+      <TestProjectName>$(MSBuildProjectName)</TestProjectName>
+    </PropertyGroup>
     <PropertyGroup>
       <TestArchiveDir>$(TestWorkingDir)$(OSPlatformConfig)/archive/tests/</TestArchiveDir>
       <ProjectJson Condition="!Exists('$(ProjectJson)')">$(OriginalProjectJson)</ProjectJson>
-    </PropertyGroup>
-    <PropertyGroup Condition="'$(TestProjectName)'==''">
-      <TestProjectName>$(MSBuildProjectName)</TestProjectName>
+      <DestinationArchiveFile>$(TestArchiveDir)$(TestProjectName)</DestinationArchiveFile>
+      <DestinationArchiveFile Condition="'$(TargetGroup)' != ''">$(DestinationArchiveFile).$(TargetGroup)</DestinationArchiveFile>
```

Comment:
I don't know enough about how this is consumed to know if there are any issues with modifying the archive name.  The logic of what you're doing looks fine to me.